apps/browser/qml/pages/components/BookmarkEditDialog.qml

```diff
@@ -83,20 +83,34 @@ UserPromptDialog {
                 id: urlField
                 text: url
 
-                acceptableInput: text.length > 0
                 onActiveFocusChanged: if (!activeFocus) errorHighlight = !acceptableInput
                 onAcceptableInputChanged: if (acceptableInput) errorHighlight = false
 
                 //: Label for textfield to edit bookmark's URL
                 //% "URL"
                 label: qsTrId("sailfish_browser-la-url_editor")
-                //% "URL is required"
-                description: errorHighlight ? qsTrId("sailfish_browser-la-url_editor_error") : ""
+                description: {
+                    if (!errorHighlight) {
+                        return ""
+                    } else if (text.length > 0) {
+                        var scheme = root.url.match(/^(https?):\/\/.+/)
+
+                        //% "URL scheme (%1) missing"
+                        return qsTrId("sailfish_browser-la-missing_url_scheme").arg(scheme.length > 1 ? scheme[1] : "https")
```

Comment:
This bit is confusing me a bit, since `scheme[1]` will only ever match `http` or `https`, and the line won't be shown in these cases. So won't this line only ever say "URL scheme (https) missing"?

Did I miss a case?

Reply:
root.url is the unedited url supplied when the dialog is shown and not the user edited one that may have had the scheme stripped from it resulting in this error. 

If there's a case when the dialog is shown without a source URL then it will always show https.

Reply:
Okay, thanks for clarifying, that makes sense.